addons/mrp/models/mrp_production.py

```diff
@@ -689,6 +689,9 @@ def _compute_date_planned_finished(self):
         for production in self:
             if not production.date_planned_start or production.is_planned:
                 continue
+            if production.date_finished:
+                production.date_planned_finished = production.date_finished
```

Comment:
It's not a good idea to forcibly set the 'date_finished' on the 'date_planned_finished' for the MO, because we want to keep a trace of the initially planned date.

When validating a manufacturing order (MO), today's date is set in the move. So, it should work correctly. So, Instead, you need to investigate why this date is being overwritten later:

https://github.com/odoo/odoo/blob/17.0/addons/mrp/models/mrp_production.py#L1984

https://github.com/odoo/odoo/blob/34c192761fa375b56d617fec78fb63d8008f6451/addons/mrp/models/mrp_production.py#L1450

https://github.com/odoo/odoo/blob/8d62eacfff5f94f692661c221c7b43f39bd3b891/addons/stock/models/stock_move.py#L1444

Reply:
the date is overridden in this code:

https://github.com/odoo/odoo/blob/7a460100716b7294ea700203c60ef0e75e359acf/addons/mrp/models/mrp_production.py#L1898-L1904

it will call this compute

https://github.com/odoo/odoo/blob/ee58488774e78e037f9b8a12082e7773949fbe9d/addons/mrp/models/mrp_production.py#L724-L737

which will set back the date to `production.date_planned_finished`

so for example adding in that method:

```diff
- if production.date_planned_finished:
-    updated_values['date'] = production.date_planned_finished
+ if production.date_finished or production.date_planned_finished:
+    updated_values['date'] = production.date_finished or production.date_planned_finished
```
or something similar should work